lib/pbench/test/unit/server/test_authorization.py

```diff
@@ -0,0 +1,144 @@
+import pytest
+
+from pbench.server.api.auth import Auth
+from pbench.server.api.resources import (
+    ApiBase,
+    API_OPERATION,
+    Schema,
+    UnauthorizedAccess,
+)
+from pbench.server.database.models.users import User
+
+
+USER_ID = "20"
+SELF_ID = "3"  # This needs to match the current_user_drb fixture
+
+
+class TestAuthorization:
+    @pytest.fixture()
+    def apibase(self, client) -> ApiBase:
+        return ApiBase(client.config, client.logger, Schema())
+
+    @pytest.fixture()
+    def current_user_admin(self, monkeypatch):
+        admin_user = User(
+            email="email@example.com",
+            id=6,
+            username="admin",
+            first_name="Test",
+            last_name="Admin",
+            role="admin",
+        )
+
+        class FakeHTTPTokenAuth:
+            def current_user(self) -> User:
+                return admin_user
+
+        with monkeypatch.context() as m:
+            m.setattr(Auth, "token_auth", FakeHTTPTokenAuth())
+            yield admin_user
+
+    @pytest.mark.parametrize(
+        "ask",
+        [
+            {"user": "drb", "access": "private", "role": API_OPERATION.UPDATE},
+            {"user": "drb", "access": "public", "role": API_OPERATION.UPDATE},
+            {"user": "drb", "access": "private", "role": API_OPERATION.DELETE},
+            {"user": "drb", "access": "public", "role": API_OPERATION.DELETE},
+            {"user": "drb", "access": "private", "role": API_OPERATION.READ},
+            {"user": "drb", "access": "public", "role": API_OPERATION.READ},
+            {"user": None, "access": "private", "role": API_OPERATION.READ},
+            {"user": None, "access": "public", "role": API_OPERATION.READ},
+        ],
+    )
+    def test_allowed_admin(self, apibase, server_config, current_user_admin, ask):
+        apibase.role = ask["role"]
+        apibase._check_authorization(ask["user"], ask["access"])
+
+    @pytest.mark.parametrize(
+        "ask",
+        [
+            {"user": None, "access": "private", "role": API_OPERATION.UPDATE},
+            {"user": None, "access": "public", "role": API_OPERATION.UPDATE},
+            {"user": None, "access": "private", "role": API_OPERATION.DELETE},
+            {"user": None, "access": "public", "role": API_OPERATION.DELETE},
+        ],
+    )
+    def test_disallowed_admin(self, apibase, server_config, current_user_admin, ask):
```

Comment:
There's no possibility of an "update all" or a "delete all" operation?

Reply:
Someday, maybe; not now. Certainly not in this PR.